### PR TITLE
acquireTokenSilent calls ssoSilent

### PR DESCRIPTION
**Pull request by tnorling**
- Move `acquireTokenByRefreshToken` to `RefreshTokenClient`
- `acquireTokenSilent` uses:
  1. `acquireCachedToken` on `SilentFlowClient`
  2. `acquireTokenByRefreshToken` on `RefreshTokenClient`
  3. `ssoSilent` if refresh token is expired
- `account` is added as optional parameter on `AuthorizationUrlRequest` to accommodate the lack of `loginHint` and `sid` on the `SilentFlowRequest` object.
- Auth Code Url parameter builder prefers `sid` over `loginHint` and will generate `login_hint` from `account.username` if neither are provided.

### CODE REVIEW

File: lib/msal-common/src/client/AuthorizationCodeClient.ts [hunk 1]
```diff
@@ -231,16 +231,17 @@ export class AuthorizationCodeClient extends BaseClient {
...
         if (request.domainHint) {
             parameterBuilder.addDomainHint(request.domainHint);
         }
 
+        // Add sid or loginHint with preference for sid -> loginHint -> username of AccountInfo object
         if (request.sid) {
             parameterBuilder.addSid(request.sid);
+        } else if (request.loginHint) {
+            parameterBuilder.addLoginHint(request.loginHint);
+        } else if (request.account && request.account.username) {
```
Comment (sameerag) [inline lib/msal-common/src/client/AuthorizationCodeClient.ts:243]: `Account` can also have `sid`? Used to in msal-core, not sure how relevant it is for browser!

Comment (jasonnutter) [reply lib/msal-common/src/client/AuthorizationCodeClient.ts:243]: We should use `sid` if it is present in the id token.

**Review by sameerag — APPROVED** [review]
One comment, approved.

File: lib/msal-common/src/client/AuthorizationCodeClient.ts (hunk 1 above)
Comment (sameerag) [reply lib/msal-common/src/client/AuthorizationCodeClient.ts:243]: Then, we need to account for `sid` in `account`

Comment (tnorling) [reply lib/msal-common/src/client/AuthorizationCodeClient.ts:243]: That's a good suggestion but will take it in a follow up PR

Comment (pkanher617) [reply lib/msal-common/src/client/AuthorizationCodeClient.ts:243]: +1 for a follow up PR

**Review by pkanher617 — APPROVED** [review]
LGTM

**Pull request merged**